chips/stm32f4xx/src/lib.rs

```diff
@@ -85,6 +85,7 @@ pub static BASE_VECTORS: [unsafe extern "C" fn(); 16] = [
 // Extracted from `CMSIS/Device/ST/STM32F4xx/Include/stm32f446xx.h`
 // NOTE: There are missing IRQn between 0 and 96
 #[cfg(feature = "stm32f446re")]
+#[cfg(not(feature = "cargo-clippy"))] //prevents error from clippy seeing two IRQS defined
```

Comment:
This seems like a bad precedent to set. Is there some sort of canonical cfg usage that avoids issues like these?

Reply:
So I poked around at this some more, and the specific reason we run into this issue is because clippy runs in parallel on all crates when clippy is called from a cargo workspace. Running `cargo clippy` in every crate individually works -- but running `cargo clippy` from the workspace runs it simultaneously on both the `nucleo_f446re` and `nucleo_f429zi` crates, meaning that *both* features are enabled when clippy runs in `chips/stm32f4xx`. I can think of 3 solutions:

1. The canonical way to handle this, I think, is to use the `cfg_if!()` crate. This crate was written by a member of the rust team and is used internally in rust libstd and libcore, but unfortunately is not exposed (https://github.com/rust-lang/rust/pull/59443). It is a just a 25 line macro, so we could copy it into Tock (I just tested this and it works).

2. We could call `cargo clippy` on each crate individually using `cargo clippy -p {CRATE}`, but then clippy will run much slower since we lose the benfits of running in parallel across all crates in the workspace.

3. A simpler fix for this specific instance would be to replace:
```rust
#[cfg(feature = "stm32f429zi")]
```
with 
```rust
#[cfg(not(feature = "stm32f446re"))]
#[cfg(feature = "stm32f429zi")]
```
or
```rust
#[cfg(all(feature = "stm32f429zi", not(feature = "stm32f446re"))]
```
  so that both definitions cannot be enabled simultaneously, but this feels like a bit more of a hack vs. using `cfg_if!()`

Thoughts?

Reply:
I vote for 1, copy it into Tock